lib/licensee/projects/uri_project.rb

```diff
@@ -0,0 +1,25 @@
+# URI-based project
+#
+# Analyzes a given URI for license information
+require 'net/http'
+require 'uri'
+
+module Licensee
+  class UriProject < Project
+    def initialize(path, **args)
+      @uri = URI(path)
+      raise UnsupportedProject if @uri.scheme !~ /^https?$/
+      super(**args)
+    end
+
+    def load_file(file)
+      Net::HTTP.get(file[:uri]) if @allow_remote
+    end
+
+    def files
```

Comment:
For other project types, `files` would return an array of files within the project. Obviously that's not possible here. What if we just made `@files` an `attr_acessor` and then a user would just do `files.push(name: "LICENSE")` with any files they knew to exist, and LICENSE can support multiple license files, multiple matching strategies, etc.

Alternatively, we could have a hard-coded list of the most common paths, call them in order, and only go to the next one if the previous doesn't exist (e.g., `LICENSE`, `LICENSE.txt`, `LICENSE.md`, etc.).